packages/viz/src/components/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,404 @@
+                  }; font-weight: ${
+            activeTheme?.fontWeights.semibold
+          }; font-size: ${activeTheme?.fontSizes.sm}; color: ${
+            activeTheme?.colors.modes[selectedMode].secondary
+          };">${tooltipName}</p>
+                </div>
+              </div>
+              <div style="display: flex; flex-direction: column; padding: ${
+                activeTheme?.space.sm
+              };">
+                ${params
+                  .map((s, i) => {
+                    return `
+                    <div key="${
+                      s.seriesName
+                    }" style="display: flex; flex-direction: row; justify-content: space-between; align-items: center; padding-bottom: ${
+                      i === params.length - 1 ? 0 : activeTheme?.space.sm
+                    };">
+                        
+                    <div style="display: flex; flex-direction: row; align-items: center; margin-right: ${
+                      activeTheme?.space.sm
+                    };">
+                          <p style="width: 10px; height: 10px; background-color: ${
+                            s.color
+                          }; margin-right: 5px"/>
+                          <p style="font-family: ${
+                            activeTheme?.fontFamily.body
+                          }; font-weight: ${
+                      activeTheme?.fontWeights.semibold
+                    }; font-size: ${activeTheme?.fontSizes.sm}; color: ${
+                      activeTheme?.colors.modes[selectedMode].secondary
+                    };">${s.seriesName}</p>
+                        </div>
+                        
+                        <p style="font-family: ${
+                          activeTheme?.fontFamily.body
+                        }; font-size: ${activeTheme?.fontSizes.sm}; color: ${
+                      activeTheme?.colors.modes[selectedMode].secondary
+                    };">${
+                      tooltip?.valueFormatter
+                        ? tooltip.valueFormatter(s.value)
+                        : s.value
+                    }</p>
+                    </div>
+                  `;
+                  })
+                  .join(" ")}
+              </div>
+            </div>
+          `;
+        },
+      },
+    };
+  }, [
+    tooltip?.show,
+    tooltip?.valueFormatter,
+    selectedMode,
+    selectedTheme,
+    activeTheme,
+  ]);
+
+  const chartLegend = useMemo(() => {
+    return {
+      legend: {
+        show: legend?.showLegend ?? chartSeries.series.length > 1,
+      },
+    };
+  }, [chartSeries, legend?.showLegend]);
+
+  const chartHorizontalRangerSlider = useMemo(() => {
+    return {
+      dataZoom:
+        horizontalRangeSlider?.show === true
+          ? {
+              type: "slider",
+              orient: "horizontal",
+            }
+          : undefined,
+    };
+  }, [horizontalRangeSlider?.show]);
+
+  return (
```

Comment:
How does the component react to changes? In particular with changes in the data?

Reply:
I tested this with the "With Interval Updates" story where the data is updating every second and the range slider was always resetting to the initial values. Because of this it was very difficult to use the range slider.

This was fixed in a new commit (I also added the changes you mentioned) and the range slider is no longer resetting the values every time the data updates. This is the current behaviour:

https://github.com/lumada-design/hv-uikit-react/assets/43220251/99b152bf-c1ce-48fb-a636-0ce793ade760